go/services/reports/pkg/scheduled_jobs/scheduled_job_repository.go

```diff
@@ -0,0 +1,175 @@
+	logger            *zap.Logger
+}
+
+func NewMongoDbScheduledJobRepository[T ScheduledJob](mongoDbCollection *repositories.MongoDbCollection[T], logger *zap.Logger) *MongoDbScheduledJobRepository[T] {
+	return &MongoDbScheduledJobRepository[T]{
+		mongoDbCollection: mongoDbCollection,
+		logger:            logger,
+	}
+}
+
+func (r *MongoDbScheduledJobRepository[T]) InsertScheduledJobs(ctx context.Context, incidents []T) ([]string, *ScheduledJobRepositoryError) {
+
+	documents := make([]interface{}, 0, len(incidents))
+
+	if len(incidents) == 0 {
+		return make([]string, 0, 0), nil
+	}
+
+	for _, incident := range incidents {
+		documents = append(documents, incident)
+	}
+
+	ids, err := r.mongoDbCollection.InsertDocuments(documents)
+
+	if err != nil {
+		r.logger.Error("Failed to insert jobs", zap.Error(err))
+		return nil, NewScheduledJobInternalError(err)
+	}
+
+	createdIds := array.Map(func(objectId primitive.ObjectID) string {
+		return objectId.Hex()
+	})(ids)
+
+	return createdIds, nil
+}
+
+func (r *MongoDbScheduledJobRepository[T]) GetScheduledJob(ctx context.Context, id string) (*T, *ScheduledJobRepositoryError) {
+
+	objectId, err := primitive.ObjectIDFromHex(id)
+	if err != nil {
+		r.logger.Error("Failed to parse application incident id", zap.Error(err))
+		return nil, NewInvalidScheduledJobIdError(err)
+	}
+
+	incident, err := r.mongoDbCollection.GetDocument(primitive.D{{Key: "_id", Value: objectId}}, primitive.D{})
+
+	if err != nil {
+		r.logger.Error("Failed to decode node incident from mongodb", zap.Error(err))
+		return nil, NewScheduledJobNotFoundError(err)
+	}
+
+	return &incident, nil
+}
+
+func (r *MongoDbScheduledJobRepository[T]) GetScheduledJobsByIds(ctx context.Context, ids []string) ([]T, *ScheduledJobRepositoryError) {
+
+	idObjects := array.Map(func(id string) primitive.ObjectID {
+		idObj, err := primitive.ObjectIDFromHex(id)
+		if err != nil {
+			r.logger.Error("Failed to parse application incident ids", zap.Error(err))
+		}
+		return idObj
+	})(ids)
+
+	incidents, err := r.mongoDbCollection.GetDocuments(bson.D{{Key: "_id", Value: bson.M{"$in": idObjects}}}, primitive.D{})
+	if err != nil {
+		r.logger.Error("Failed to find application incidents by ids", zap.Error(err))
+		return nil, NewScheduledJobInternalError(err)
+	}
+
+	return incidents, nil
+}
+
+func (r *MongoDbScheduledJobRepository[T]) GetScheduledJobsByStatus(ctx context.Context, status string) ([]T, *ScheduledJobRepositoryError) {
+
+	incidents, err := r.mongoDbCollection.GetDocuments(bson.D{{Key: "status", Value: status}}, bson.D{{Key: "scheduledAt", Value: 1}})
+	if err != nil {
+		r.logger.Error("Failed to find application incidents by status", zap.Error(err), zap.Any("status", status))
+		return nil, NewScheduledJobInternalError(err)
+	}
+
+	return incidents, nil
+}
+
+func (r *MongoDbScheduledJobRepository[T]) UpdateScheduledJob(ctx context.Context, scheduledJob T) *ScheduledJobRepositoryError {
```

Comment:
Update operation changes id of the document, that is kind of inconvenient. Maybe the function can be renamed to ReplaceScheduledJob or the id can be preserved?